sc-web/components/scgg/src/scgg-struct.js

```diff
@@ -256,112 +250,250 @@ function scggScStructTranslator(_editor, _sandbox) {
+                                            sc_type_arc_pos_const_perm,
+                                            SCggKeynodesHandler.scKeynodes.rrel_current_version
+                                        ]).done(function(results){ 
+                                            SCWeb.core.Server.resolveScAddr(['sc_garbage'], function (keynodes) {     
+                                                window.sctpClient.create_arc(sc_type_arc_pos_const_perm, keynodes.sc_garbage, results[0][3])
+                                                .done(dfdDeleteCurrent.resolve)
+                                                .fail(dfdDeleteCurrent.reject);
+                                            });
+                                            window.sctpClient.erase_element(results[0][2]).done(dfdDeleteCurrent.resolve)
+                                            .fail(dfdDeleteCurrent.reject);
+                                        }).fail(dfdDeleteCurrent.resolve);
+
+                                    return dfdDeleteCurrent.promise();
+                                }
+
+                                var addCurrent = function(){
+                                    var addCurrent = new jQuery.Deferred();
+
+                                        window.sctpClient.create_arc(sc_type_arc_pos_const_perm, editor.render.sandbox.decompositionNodeAddr, addrStruct).done(function (arcSysIdtf){
+                                            window.sctpClient.create_arc(sc_type_arc_pos_const_perm, SCggKeynodesHandler.scKeynodes.rrel_current_version, arcSysIdtf)
+                                            .done(function(){
+                                                addCurrent.resolve();
+                                            }).fail(addCurrent.reject);        
+                                        }).fail(addCurrent.reject);
+
+                                    return addCurrent.promise();
+                                }
+
+                                deleteCurrent().always(function(){
+                                    addCurrent()
+                                    .done(dfdAddCurrentGraph.resolve)
+                                    .fail(dfdAddCurrentGraph.reject);
+                                });
+
+                                return dfdAddCurrentGraph.promise();
+                            };
+
+                           
+                       // findOrCreateDecomposition().done(dfdTranslateDecomposition.resolve);
+                        if (!editor.render.sandbox.loadGraph){
+                        createDecomposition().done(function() {
+                            addCurrentGraph().done(dfdTranslateDecomposition.resolve)
+                            .fail(dfdTranslateDecomposition.reject);
+                        }).fail(dfdTranslateDecomposition.reject);
+                       }
+                        else{
+                        if (editor.render.sandbox.graphNodeAddr === null){
```

Comment:
if editor.render.sandbox.graphNodeAddr === null
then editor.render.sandbox.decompositionNodeAddr null to and you need create main graph node and decompositionNode